services/QuillLMS/client/app/bundles/Shared/utils/stringFormatting.ts

```diff
@@ -0,0 +1,15 @@
+
+class StringFormatting {
```

Comment:
I'd like to move toward expressing static utility functions as class methods. It allows use to put functions in logical groups, and also adds namespacing and signals intent.

Reply:
I like the idea of this extraction.  Just a thought, but what about going a step further and renaming the function name the function name to  `StringFormatting.highlightSpellingGrammar(...)`  since the class is already expressing that this is formatting?

Reply:
I'm not strictly against it in this context, but I do think that in general, we are moving towards a functional rather than OO approach in our Javascript and it probably doesn't make sense to backtrack on that here. I think file structure can group things logically without the use of classes.

Reply:
Also, I think that since this function is used in such a specific capacity, it's better to keep it closer in the file structure to where it is actually employed (in only one place, right now) than all the way out in `Shared/utils`. Curious if @eadams17 has thoughts about either of these points.

Reply:
I agree with @emilia-friedberg. In general, React is moving away from the use of classes and is encouraging functional development with its newer features so I think we want to continue on that course. I also agree on the second point; there is already a `/libs` directory within the Comprehension app where I think makes more sense to house this since it's specific to Comprehension